metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java

```diff
@@ -1497,16 +1500,20 @@ public static boolean isExternalTable(Table table) {
     if (table == null) {
       return false;
     }
-    String tableType = table.getTableType();
-    if (tableType != null && tableType.equals(TableType.EXTERNAL_TABLE.name())) {
-      return true;
-    }
+
     Map<String, String> params = table.getParameters();
     if (params == null) {
-      return false;
+      return isTableTypeExternal(table);
+    }
+    else {
+      String externalParam = params.get(EXTERNAL_TABLE_PROP);
+      if (externalParam != null) {
+        return Boolean.parseBoolean(params.get(EXTERNAL_TABLE_PROP));
```

Comment:
Nit: swap out  `params.get(EXTERNAL_TABLE_PROP)` for `externalParam`?

Reply:
Nit 2: what about having flat `if-case` here? 

```
    if (params == null) {
        ...
    } else if (params.get(EXTERNAL_TABLE_PROP) == null ) {
        ...
    } else ...

Reply:
Ok for first comment. For the second if I actually follow the previous style it would be more sth like this:

```
if (params == null) {
  return xxx
}

String externalParam = params.get(EXTERNAL_TABLE_PROP);
if (externalParams != null) {
  return yyy;
}

return zzz;
```

In any case, both cannot be done at the same time